pkg/commands/workload.go

```diff
@@ -155,6 +157,9 @@ func (opts *WorkloadOptions) Validate(ctx context.Context) validation.FieldError
 	if opts.SourceImage != "" {
 		source = append(source, flags.SourceImageFlagName)
 	}
+	if opts.SubPath != "" && len(source) != 1 {
+		errs = errs.Also(validation.ErrDisallowedFields(flags.SubPathFlagName, "subPath flag cannot be used without local-path/git-* sources"))
+	}
```

Comment:
This check needs to be lazy. It should be valid to specify `--subpath` for an update when there is already source defined, or for a create with a file where the workload already has source defined.

Reply:
+1